File: src/main/java/com/sistemalima/casadocodigodeveficiente/repository/author/AuthorRepository.java
package com.sistemalima.casadocodigodeveficiente.repository.author;

import com.sistemalima.casadocodigodeveficiente.repository.author.entity.AuthorEntity;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface AuthorRepository extends CrudRepository<AuthorEntity, Long> {
    Optional<AuthorEntity> findByEmail(String email);
}

File: src/main/kotlin/com/sistemalima/casadocodigodeveficiente/repository/AuthorRepository.kt
package com.sistemalima.casadocodigodeveficiente.repository

import com.sistemalima.casadocodigodeveficiente.repository.entity.AuthorEntity
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface AuthorRepository: CrudRepository<AuthorEntity, Long> {

    fun findByEmail(email: String): Optional<AuthorEntity>
}